packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/categories/UpdateCategoriesTests.java

```diff
@@ -0,0 +1,235 @@
+        restClient.authenticateUser(dataUser.getAdminUser())
+            .withCoreAPI()
+            .usingCategory(categoryWithFolderId)
+            .updateCategory(fixedCategoryModel);
+
+        restClient.assertStatusCodeIs(BAD_REQUEST);
+    }
+
+    /**
+     * Try to update a root category and receive 400 (Bad Request)
+     */
+    @Test(groups = { TestGroup.REST_API})
+    public void testUpdateCategory_usingRootCategoryAndExpect400()
+    {
+        STEP("Create root category model");
+        final RestCategoryModel rootCategoryModel = createCategoryModelWithId(ROOT_CATEGORY_ID);
+
+        STEP("Try to update as admin root category");
+        final RestCategoryModel fixedCategoryModel = createCategoryModelOf(getRandomName("NewCategoryName"), rootCategoryModel);
+        restClient.authenticateUser(dataUser.getAdminUser())
+            .withCoreAPI()
+            .usingCategory(rootCategoryModel)
+            .updateCategory(fixedCategoryModel);
+
+        restClient.assertStatusCodeIs(BAD_REQUEST);
+    }
+
+    /**
+     * Try to update a category with an empty name and receive 400 (Bad Request)
+     */
+    @Test(groups = { TestGroup.REST_API})
+    public void testUpdateCategory_withEmptyNameAndExpect400()
+    {
+        STEP("Prepare as admin a category under root category");
+        final RestCategoryModel createdCategory = prepareCategoryUnderRoot();
+
+        STEP("Try to update as admin newly created category with a category without name");
+        final RestCategoryModel fixedCategoryModel = createCategoryModelOf(null, createdCategory);
+        restClient.authenticateUser(dataUser.getAdminUser())
+            .withCoreAPI()
+            .usingCategory(createdCategory)
+            .updateCategory(fixedCategoryModel);
+
+        restClient.assertStatusCodeIs(BAD_REQUEST);
+    }
+
+    /**
+     * Try to update a category using not matching ID in path and receive 400 (Bad Request)
+     */
+    @Test(groups = { TestGroup.REST_API})
+    public void testUpdateCategory_withNotMatchingIdInPathAndExpect400()
+    {
+        STEP("Prepare as admin two categories under root category");
+        final RestCategoryModel createdCategory = prepareCategoryUnderRoot();
+        final RestCategoryModel notRelatedCategory = prepareCategoryUnderRoot();
+
+        STEP("Try to update as admin newly created category using wrong ID in path");
+        final RestCategoryModel fixedCategoryModel = createCategoryModelOf(getRandomName("NewCategoryName"), createdCategory);
+        restClient.authenticateUser(dataUser.getAdminUser())
+            .withCoreAPI()
+            .usingCategory(notRelatedCategory)
+            .updateCategory(fixedCategoryModel);
+
+        restClient.assertStatusCodeIs(BAD_REQUEST);
+    }
+
+    /**
+     * Try to update a category using not matching ID in request body and receive 400 (Bad Request)
```

Comment:
I'm pretty sure that we shouldn't validate the id in the body. It should be allowed to be there, but get ignored. I'm pretty sure this is the behaviour for other PUT endpoints.